pkg/controller/appdefinition/pullappimage.go

```diff
@@ -58,12 +58,16 @@ func pullAppImage(transport http.RoundTripper, client pullClient) router.Handler
 		var (
 			resolved string
 			err      error
+			isLocal  bool
 		)
 		if !appInstance.Status.AvailableAppImageRemote {
-			resolved, _, err = client.resolve(req.Ctx, req.Client, appInstance.Namespace, target)
+			resolved, isLocal, err = client.resolve(req.Ctx, req.Client, appInstance.Namespace, target)
 			if err != nil {
 				cond.Error(err)
 				return nil
+			} else if !isLocal {
+				// force pull from remote, since the only local image we found was marked remote, and there might be a newer version
```

Comment:
nit: I haven't been able to track down explicit documentation about this, but I'm fairly sure the rule-of-thumb for Go commentary -- except above declarations -- is to capitalize the first letter; at least, every major Go project I've seen seems to follow that convention (e.g. [k8s](https://github.com/kubernetes/kubernetes/blob/b63700630231803967f85e83ff35d719db511baa/pkg/controller/deployment/sync.go#L55))

```suggestion
				// Force pull from remote, since the only local image we found was marked remote, and there might be a newer version
```